### ISSUE TITLE
Set-Cookie header not working properly when redirecting

### ISSUE THREAD

**Issue by TelephoneTan** [getinsomnia/insomnia#6478]
### Expected Behavior

when redirection happens, the subsequent Set-Cookie response header set cookies to the subsequent url domains

### Actual Behavior

when redirection happens, the subsequent Set-Cookie response header set cookies to the original url domains

### Reproduction Steps

trigger a request with 302 status code and Set-Cookie response header

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Insomnia Version

2023.5.8

### What operating system are you using?

Windows

### Operating System Version

版本	Windows 11 专业版 版本	22H2 安装日期	‎2022/‎5/‎12 操作系统版本	22624.2265 体验	Windows Feature Experience Pack 1000.22674.1000.0

### Installation method

download from insomnia.rest

### Last Known Working Insomnia version

2021.6.0

**Comment by TelephoneTan** [comment]
anyone here?

**Comment by jrmcdonald** [comment]
I am also experiencing this issue on Ubuntu 22.04.3 LTS with version 8.3.0.

**Comment by git-commit-amen** [comment]
Correct me if I am reading this wrong but just to clarify, are you reporting that when sending a request that responds with a 3xx code and the following headers:

```
Location: otherdomain.com
Set-Cookie: foo=bar
```

The `foo=bar` cookie gets set for the domain you originally sent the request to, instead of the cookie getting set for *otherdomain.com*?

If my understanding is correct, this is probably not a bug. Browsers and HTTP clients (which support cookies) do not allow sites to do this because if they did, any website would be able to easily erase or update any other website's cookies simply by returning a redirect with a Set-Cookie header.

If you want Insomnia to do this, it sounds like something that might be better as a feature request - perhaps in the form of an optional setting.

Let me know if this makes sense. Thanks!

**Comment by TelephoneTan** [comment]
> Correct me if I am reading this wrong but just to clarify, are you reporting that when sending a request that responds with a 3xx code and the following headers:
> 
> ```
> Location: otherdomain.com
> Set-Cookie: foo=bar
> ```
> 
> The `foo=bar` cookie gets set for the domain you originally sent the request to, instead of the cookie getting set for *otherdomain.com*?
> 
> If my understanding is correct, this is probably not a bug. Browsers and HTTP clients (which support cookies) do not allow sites to do this because if they did, any website would be able to easily erase or update any other website's cookies simply by returning a redirect with a Set-Cookie header.
> 
> If you want Insomnia to do this, it sounds like something that might be better as a feature request - perhaps in the form of an optional setting.
> 
> Let me know if this makes sense. Thanks!

No, Insomnia follows redirects by default, after it sends request to `otherdomain.com`, a Set-Cookie header  is returned by the server, but Insomnia stores this cookie as the `originaldomain.com`'s cookie.

**Comment by git-commit-amen** [comment]
Ah ok apologies - I misunderstood that. I thought you meant that the _first_ server was the one returning the `Set-Cookie` header.

That sounds like a bug in this case.

**Comment by sevenam** [comment]
Think I am seeing this issue as well. Was using an old 2023 version of Insomnia, where setting the cookie worked fine. My colleague couldn't get it to work in v11.4.0. I fetched this version and experienced the same problem. Could not get it to work in this version... Tried Postman as well, which also worked fine.
